test/integration/driver.go

```diff
@@ -232,28 +231,34 @@ func getRestartEpoch(pod string) (int, error) {
 	return 0, fmt.Errorf("could not obtain envoy restart epoch")
 }
 
-func addConfig(config []byte, kind, name string) {
+func addConfig(config []byte, kind, name string, create bool) {
 	log.Println("Add config")
 	log.Println(string(config))
-	out, err := yaml.YAMLToJSON(config)
-	check(err)
 	istioKind, ok := model.IstioConfig[kind]
 	if !ok {
 		check(fmt.Errorf("Invalid kind %s", kind))
 	}
-	v, err := istioKind.FromJSON(string(out))
+	v, err := istioKind.FromYAML(string(config))
 	check(err)
-	check(istioClient.Put(model.Key{
+	key := model.Key{
 		Kind:      kind,
 		Name:      name,
 		Namespace: params.namespace,
-	}, v))
+	}
+	if create {
+		check(istioClient.Post(key, v))
+	} else {
+		check(istioClient.Put(key, v))
+	}
 }
```

Comment:
I wonder if instead of having to CLI commands (POST and PUT), there could just be one (ADD) with "add or create" semantics? Or all of the above, but maybe that's overkill.